recipe/meta.yaml

```
@@ -50,8 +50,7 @@ requirements:
     - patchelf <0.18     # [linux]
     - pkginfo
     - psutil
-    # https://github.com/conda/conda-build/issues/5594
-    - py-lief <0.15
+    - py-lief
```

```suggestion
    - py-lief <0.18
```

conda-build just merged a PR with this bound. We should probably keep this for now.

---

If we want to reduce breakage due to future versions, we should do `< 0.17`. As the lastest lief [release](https://github.com/lief-project/LIEF/releases/) is 0.16.3, it would be pure speculation to claim that conda-build is compatible with 0.17 (and/or incompatible with 0.18)

---

Or perhaps you're confusing this with patchelf, where the bound actually is `0.18`? 🤔 

If so, we've already got that bound here
https://github.com/conda-forge/conda-build-feedstock/blob/fd8f5d5d9aa1e8f86bfe0eb4f2f64fdb930fb9c7/recipe/meta.yaml#L50
In fact, https://github.com/conda/conda-build/pull/5607 came after https://github.com/conda-forge/conda-build-feedstock/commit/3d043fe301ad48ba2025f7a5092f7512d5313971

---

Ah woops I indeed confused that. 0.17 is good by me.

---

I disagree with putting a bound on a release that doesn't exist yet. There may be a breaking change that's relevant here, there might not. We don't know yet. Unless you're aware of something relevant that's already been merged to main of lief.

---

We've had enough breakage with recent pylief releases that I think it's defensible to add this cap, and lift it once a new version exists and has been tested to be compatible.

---

> There were no new breaking changes in 0.16 (for conda-build's purposes) that weren't already in 0.15.

Which is why 0.16 isn't being excluded here.

To be clear, I'm at best lukewarm on the cap, but if it's @beckermr's preference, that's fine by me.

> But we've also seen that conda-build's tests are not covering the codepaths for this dependency very well on CI.

If you care about removing the cap, it would be an excellent first step to add tests to upstream conda-build that would have caught https://github.com/conda/conda-build/issues/5564 & https://github.com/conda/conda-build/issues/5594

---

I'd have loved to include test cases in my PR's that fixed those issues, if I understood why real-world usage of conda-build was exhibiting the failures but the existing tests didn't. Would be more appropriate to have that conversation in a new issue there.